user_guide_src/source/models/model.rst

```diff
@@ -311,13 +311,15 @@ Saving Data
 insert()
 --------
 
-An associative array of data is passed into this method as the only parameter to create a new
-row of data in the database. The array's keys must match the name of the columns in a ``$table``, while
-the array's values are the values to save for that key:
+The first parametre is an associative array of data to create a new row of data in the database. 
+If an object is passed instead of an array, it will attempt to convert it to an array.
+
+The array's keys must match the name of the columns in the ```$table```, while the array's values are the values to save for that key:
 
 .. literalinclude:: model/015.php
 
-You can retrieve the last inserted row's primary key using the ``getInsertID()`` method.
```

Comment:
Do not delete the description for ``getInsertID()``. The method is only here.

Reply:
This results in an error: Call to a member function getInsertID() on bool.

`echo $this->users->insert($data, false)->getInsertID();`

The way to get the insert ID is to omit the second parameter or if its `false` use the `getInsertID()` on the model:

`$this->users->getInsertID();`

Which is not related to the `insert()` method. So do you think having details related to the `getInsertID()` method under the `insert()` to be out of context?

Should it just have its own section like `insert()`, `update()`, `getInsertID()` ...etc?

Reply:
`getInsertID()` is related to only insert. So it should be under `insert()` section.

Reply:
As far as I can tell it is used `$this->model->getInsertID();` not `$this->model->insert($data)->getInsertID();` so should a more detailed paragraph under the `insert()` be created then if its too small to be in its own section?

Reply:
I agree with kenjis. There is no need for a detailed paragraph because the `getInsertID()` method is only usable in the context of the `insert()` method.

The only thing that might be missing is a simple example + description, which should clear up any doubts.